wcfsetup/install/files/ts/WoltLabSuite/Core/Media/Manager/Search.ts

```diff
@@ -0,0 +1,183 @@
+    });
+
+    this._mediaManager.getDialog().querySelector(".dialogContent")!.scrollTop = 0;
+  }
+
+  /**
+   * Cancels the search after clicking on the cancel search button.
+   */
+  protected _cancelSearch(): void {
+    if (this._searchMode) {
+      this._searchMode = false;
+
+      this.resetSearch();
+      this._mediaManager.resetMedia();
+    }
+  }
+
+  /**
+   * Hides the search string threshold error.
+   */
+  protected _hideStringThresholdError(): void {
+    const innerInfo = DomTraverse.childByClass(
+      this._input.parentNode!.parentNode as HTMLElement,
+      "innerInfo",
+    ) as HTMLElement;
+    if (innerInfo) {
+      innerInfo.style.display = "none";
+    }
+  }
+
+  /**
+   * Handles the `[ENTER]` key to submit the form.
+   */
+  protected _keyPress(event: KeyboardEvent): void {
+    if (event.key === "Enter") {
+      event.preventDefault();
+
+      if (this._input.value.length >= this._mediaManager.getOption("minSearchLength")) {
+        this._hideStringThresholdError();
+
+        this.search();
+      } else {
+        this._showStringThresholdError();
+      }
+    }
+  }
+
+  /**
+   * Shows the search string threshold error.
+   */
+  protected _showStringThresholdError(): void {
+    let innerInfo = DomTraverse.childByClass(
+      this._input.parentNode!.parentNode as HTMLElement,
+      "innerInfo",
+    ) as HTMLParagraphElement;
+    if (innerInfo) {
+      innerInfo.style.display = "block";
+    } else {
+      innerInfo = document.createElement("p");
+      innerInfo.className = "innerInfo";
+      innerInfo.textContent = Language.get("wcf.media.search.info.searchStringThreshold", {
+        minSearchLength: this._mediaManager.getOption("minSearchLength"),
+      });
+
+      (this._input.parentNode! as HTMLElement).insertAdjacentElement("afterend", innerInfo);
+    }
+  }
+
+  /**
+   * Hides the media search.
+   */
+  public hideSearch(): void {
+    this._searchContainer.style.display = "none";
+  }
+
+  /**
+   * Resets the media search.
+   */
+  public resetSearch(): void {
+    this._input.value = "";
+  }
+
+  /**
+   * Shows the media search.
+   */
+  public showSearch(): void {
+    this._searchContainer.style.display = "block";
+  }
+
+  /**
+   * Sends an AJAX request to fetch search results.
+   */
+  public search(pageNo?: number): void {
+    if (typeof pageNo !== "number") {
+      pageNo = 1;
+    }
+
+    let searchString = this._input.value;
+    if (searchString && this._input.value.length < this._mediaManager.getOption("minSearchLength")) {
+      this._showStringThresholdError();
+
+      searchString = "";
+    } else {
+      this._hideStringThresholdError();
+    }
+
+    this._searchMode = true;
+
+    Ajax.api(this, {
+      parameters: {
+        categoryID: this._mediaManager.getCategoryId(),
+        imagesOnly: this._mediaManager.getOption("imagesOnly"),
+        mode: this._mediaManager.getMode(),
+        pageNo: pageNo,
+        searchString: searchString,
```

Comment:
These _could_ simply be:
```suggestion
        pageNo,
        searchString,
```